### ISSUE TITLE
Support container-based development workflow

### ISSUE THREAD

**Issue by manning-ncsa** [USRSE/usrse23#7]
[The current Dockerfile](https://github.com/USRSE/usrse23/blob/1e5a54fa96fd698cc89d8c9059dcc6f2fc1c494d/Dockerfile) fails to build.  Update the Dockerfile and related files to enable reproducible builds of the website and support local container-based development workflows conducive to code collaboration.

**Comment by jbteves** [comment]
@manning-ncsa please paste the command run to build the dockerfile and the subsequent error message you receive.

**Comment by manning-ncsa** [comment]
I added the `faraday-retry` gem to the Gemfile due to a suggestion that displayed during the bundle install step when it was absent.

**Comment by mrmundt** [comment]
For posterity, the issue with the Gemfile was that it was an old copy. **_The right copy is in the PR attached to this issue_**. This was the error:

```
Fetching gem metadata from https://rubygems.org/...........
Resolving dependencies...
Bundler found conflicting requirements for the Ruby version:
  In Gemfile:
    github-pages (= 193) was resolved to 193, which depends on
      listen (= 3.1.5) was resolved to 3.1.5, which depends on
        Ruby (>= 2.2.3, ~> 2.2)

  Current Ruby version:
    Ruby (= 3.1.3)

Bundler could not find compatible versions for gem "rouge":
  In Gemfile:
    github-pages (= 193) was resolved to 193, which depends on
      rouge (= 2.2.1)

    github-pages (= 193) was resolved to 193, which depends on
      jekyll (= 3.7.4) was resolved to 3.7.4, which depends on
        rouge (< 4, >= 1.7)
```
